tests/integration_tests/functional/test_error_code.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests scenarios for Firecracker signal handling."""
+
+import os
+import time
+import platform
+
+def test_enosys_error_code(test_microvm_with_ssh, network_config):
+    """Test that handled signals don't kill the microVM."""
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+
+    # We don't need to monitor the memory for this test.
+    test_microvm.memory_monitor = None
+
+    test_microvm.kernel_file = test_microvm.kernel_file[:-11] + "enosys_vmlinux.bin"
+    test_microvm.basic_config(vcpu_count=1)
+
+    test_microvm.start()
+
+    firecracker_pid = test_microvm.jailer_clone_pid
+
+    while True:
+        # Pytest's timeout will kill the test even if the loop doesn't exit.
+        try:
+            os.kill(firecracker_pid, 0)
+            time.sleep(0.01)
+        except OSError:
+            break
+
+    if platform.machine() != "x86_64":
+        log_data = test_microvm.log_data
+        assert "Received ENOSYS error because KVM failed to emulate an instruction" in log_data
+        assert "Vmm is stopping." in log_data
```

Comment:
This integration test is implemented here and the next commit adds changes to it. Shouldn't it be better to implement & manage it in only one commit?